collector/network.go

```diff
@@ -100,9 +95,72 @@ func (c *NetworkCollector) Collect(ch chan<- prometheus.Metric) {
 		connStats[direction][private]++
 	}
 
-	for direction, stats := range connStats {
-		for private, count := range stats {
-			ch <- prometheus.MustNewConstMetric(connsDesc, prometheus.GaugeValue, float64(count), direction, private)
+	return connStats, nil
+}
+
+func (c *NetworkCollector) getBootstrapped(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, bootstrappedTimeout)
+	defer cancel()
+
+	ch := make(chan *tezos.BootstrappedBlock, 10)
+	var err error
+
+	go func() {
+		err = c.service.GetBootstrapped(ctx, ch)
+		close(ch)
+	}()
+
+	var cnt int
+	for range ch {
+		if cnt > 0 {
+			// More than one record returned
+			return false, nil
+		}
+		cnt++
+	}
+
+	if err != nil {
+		if e, ok := err.(net.Error); ok && e.Timeout() {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+// Collect implements prometheus.Collector and is called by the Prometheus registry when collecting metrics.
+func (c *NetworkCollector) Collect(ch chan<- prometheus.Metric) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+
+	if stats, err := c.service.GetNetworkStats(ctx); err != nil {
+		c.errors.WithLabelValues("network_stat").Add(1)
+		level.Warn(c.logger).Log("msg", "error querying /network/stat", "err", err)
+	} else {
+		ch <- prometheus.MustNewConstMetric(sentBytesDesc, prometheus.CounterValue, float64(stats.TotalBytesSent))
+		ch <- prometheus.MustNewConstMetric(recvBytesDesc, prometheus.CounterValue, float64(stats.TotalBytesRecv))
+	}
+
+	if connStats, err := c.getConnStats(ctx); err != nil {
+		c.errors.WithLabelValues("network_connections").Add(1)
```

Comment:
These label values don't match up with the initial set to `0` for the label value `network` in the constructor anymore, so there's going to be a `0` count sticking around for `network` forever and the other label values aren't initialized to `0` at all.

Reply:
Actually from a monitoring perspective, it would be more useful to have a 0/1 gauge for every scrape (with the same dimensions) rather than a counter. Couple of reasons:

- There can't be multiple increases between scrapes that need to be added up, since the errors only occur during scrapes (even with parallel scrapes the semantic result of each Prom knowing the success of its own scrapes is fine).
- For alerting, it's more robust to alert on certain collectors failing consistently for a while instead of defining a particular rate at which they may fail (which will depend on scrape rate and 
the number of parallelly scraping Prometheus servers).